packages/@aws-cdk/core/lib/expiration.ts

```diff
@@ -0,0 +1,69 @@
+import { Duration } from './duration';
+/**
+ * Represents a date of expiration.
+ *
+ * The amount can be specified either as a Date object, timestamp, Duration or string.
+ */
+export class Expiration {
+  /**
+   * Expire at the specified date
+   * @param d date to expire at
+   */
+  public static atDate(d: Date) { return new Expiration(d.toUTCString(), d); }
+
+  /**
+   * Expire at the specified timestamp
+   * @param t timestamp in unix milliseconds
+   */
+  public static atTimestamp(t: number) { return Expiration.atDate(new Date(t)); }
+
+  /**
+   * Expire once the specified duration has passed since deployment time
+   * @param t the duration to wait before expiring
+   */
+  public static after(t: Duration) { return Expiration.atDate(new Date(Date.now() + t.toMilliseconds())); }
+
+  /**
+   * Expire at specified date, represented as a string
+   *
+   * @param s the string that represents date to expire at
+   */
+  public static fromString(s: string) { return new Expiration(s, new Date(s)); }
+
+  /**
+   * Expiration value as a string
+   */
+  public readonly value: string;
```

Comment:
Why is this needed? Seems like this the UTC representation of the date, which can be obtained from `x.date.toUTCString()`, no?

Reply:
s3-deployments uses the `value` property, I added the `date` property myself, but I wouldn't mind just removing the `date` property entirely.